content/src/denylist/DenylistFactory.ts

```diff
@@ -1,9 +1,15 @@
 import { Bean, Environment, EnvironmentConfig } from '../Environment'
+import { ActiveDenylist } from './ActiveDenylist'
 import { Denylist } from './Denylist'
+import { NoopDenylist } from './NoopDenylist'
 
 export class DenylistFactory {
   static create(env: Environment): Denylist {
-    return new Denylist(
+    if (env.getConfig(EnvironmentConfig.DISABLE_DENYLIST)) {
```

Comment:
Could you remove the logic that is in the `ControllerFactory` that used to do this? It makes no sense to have both